hadoop-mapreduce-project/hadoop-mapreduce-client/hadoop-mapreduce-client-core/src/main/java/org/apache/hadoop/mapred/lib/MultipleOutputs.java

```diff
@@ -528,8 +545,33 @@ public void collect(Object key, Object value) throws IOException {
    *                             could not be closed properly.
    */
   public void close() throws IOException {
+    int nThreads = 10;
+    AtomicReference<IOException> ioException = new AtomicReference<>();
+    ExecutorService executorService = Executors.newFixedThreadPool(nThreads);
```

Comment:
I recommend using the version of this method that accepts a `ThreadFactory`, and probably use `ThreadFactoryBuilder`. The factory should generate threads that 1) use a naming format that makes it clear these threads are related to the closing process (e.g. "MultipleOutputs-close"), and 2) set an `UncaughtExceptionHandler` that logs the exception, which would make visible unexpected errors like unchecked exceptions.

Reply:
Done